packages/dart_frog/lib/src/request.dart

```diff
@@ -115,11 +117,25 @@ class Request {
     return HttpMethod.values.firstWhere((m) => m.value == _request.method);
   }
 
+  Future<List<int>> _bytes() async {
+    if (_bodyBytesCompleter == null) {
+      _bodyBytesCompleter = Completer<List<int>>();
+      final bytes = await _request.read().fold<List<int>>(
+        <int>[],
+        (previous, element) => previous..addAll(element),
+      );
+      _bodyBytesCompleter!.complete(bytes);
+    }
+    return _bodyBytesCompleter!.future;
+  }
+
   /// Returns a [Stream] representing the body.
-  Stream<List<int>> bytes() => _request.read();
+  Stream<List<int>> bytes() async* {
```

Comment:
that's the one I wouldn't apply any cache, because if you have 10Gb of data together it is unlikely you want it cached in memory. You'll pipe to filesystem and that's it.